synfig-studio/src/synfigapp/actions/valuedescbonesetparent.cpp

```diff
@@ -154,6 +154,10 @@ Action::ValueDescBoneSetParent::prepare()
 
 	if(child_bone){
 		if(auto new_parent_bone_valuenode = ValueNode_Bone::Handle::cast_dynamic(value_desc.get_parent_value_node())){
+			ValueNode_Bone::BoneSet possible_parents = child_bone->get_possible_parent_bones(child_bone);
+			if (possible_parents.count(new_parent_bone_valuenode) <= 0) {
```

Comment:
> A bone cannot set its child as its parent.
> Or maybe we should swap in this case?

@morevnaproject Any ideas here?

Reply:
Do you suggest to swap parent and child? I think this behavior will be non-intuitive, because the action is named "Set as Parent to Active Bone" - https://github.com/synfig/synfig/pull/2465/files#diff-f5abcebec5c258b3fc2e32b68dfde72926ba2decf7c02dfd61574a7393620a98R51

Reply:
So the only thing left is to fix for skeleton deformation

Reply:
Done! Happy New Year! :D